ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsMetering.scala

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend
+
+import com.daml.platform.store.backend.MeteringStorageBackend.TransactionMetering
+import org.scalatest.Inside
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+private[backend] trait StorageBackendTestsMetering
+    extends Matchers
+    with Inside
+    with StorageBackendSpec {
+  this: AsyncFlatSpec =>
+
+  behavior of "StorageBackend (metering)"
+
+  import StorageBackendTestValues._
+
+  it should "persist transaction metering" in {
+
+    val now = someTime
+    val toOffset = offset(5)
+
+    val metering = TransactionMetering(
+      someApplicationId,
+      actionCount = 1,
+      fromTimestamp = now.addMicros(2),
+      toTimestamp = now.addMicros(3),
+      fromLedgerOffset = offset(4),
+      toLedgerOffset = toOffset,
+    )
+    val expected = List(metering)
+    for {
+      _ <- executeSql(backend.parameter.initializeParameters(someIdentityParams))
+      _ <- executeSql(ingest(Vector(dtoTransactionMetering(metering)), _))
```

Comment:
Could you add a ttest with multiple metering records?

Reply:
There are multiple metering records, one for offsets 1-5, the other for offsets 6-10.